### PR TITLE
TEST/UCT: Fix test_pending with ooo.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by yosefe** [comment]
@shssf pls review

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1348/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3294/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1351/ for details.

File: test/gtest/uct/uct_test.h [hunk 1]
```diff
@@ -155,15 +155,29 @@ class uct_test : public testing::TestWithParam<const resource*>,
...
         while ((ucs_get_time() < deadline) && (!(*flag))) {
             short_progress_loop();
         }
     }
 
+    template <typename T>
+    void wait_for_value(volatile T *var, T value, bool progress,
+                        double timeout = DEFAULT_TIMEOUT_SEC) const
+    {
+        ucs_time_t deadline = ucs_get_time() + ucs_time_from_sec(timeout);
```
Comment (shssf) [inline test/gtest/uct/uct_test.h]: Perhaps we need to take the VALGRIND into account

**Comment by shssf** [comment]
:+1:

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1355/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3300/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1358/ for details.

**Pull request merged**